doc/adr/0009-time-graph-search-functionality.md

```diff
@@ -0,0 +1,154 @@
+```
+{
+  "parameters": {
+    "requested_timerange": {
+      "start": 111111111,
+      "end": 222222222,
+      "nbTimes": 1920
+    },
+    "requested_items": [
+      1,
+      2
+    ],
+    "filter_query_parameters": {
+	  "strategy": "deep",
+	  "filter_expressions_map": {    
+        "1": ["openat"],	   
+	  }
+    },
+  }
+}
+```
+Here are the following patches for implementing time graph filter capability through `filter_query_parameters`:
+-   Update query parameters for `/states` in Trace Compass Incubator: [https://git.eclipse.org/r/c/tracecompass.incubator/org.eclipse.tracecompass.incubator/+/196610](https://git.eclipse.org/r/c/tracecompass.incubator/org.eclipse.tracecompass.incubator/+/196610)
+-   Updates to swagger documentation: [https://git.eclipse.org/r/c/tracecompass.incubator/org.eclipse.tracecompass.incubator/+/196592](https://git.eclipse.org/r/c/tracecompass.incubator/org.eclipse.tracecompass.incubator/+/196592)
+-   Updates to the TSP documentation: [https://github.com/eclipse-cdt-cloud/trace-server-protocol/pull/84](https://github.com/eclipse-cdt-cloud/trace-server-protocol/pull/84)
+
+In the previous examples we noticed that the array of search expressions takes a key value of "1” (means DIMMED) or “4” (means EXCLUDED). We should also discuss the possibility of explicitly mentioning "DIMMED" or "EXCLUDED".
+
+We also need to take a look at the response for `/states` when we add search/filter expressions in our request. The response body would still be a list of time graph entry states for each row. Every state has a bit-mask called `tags`, which helps indicate whether it matched the search/filter expression or not. For example, if we ask for a 'search' expression the states that don't match it would be dimmed, and the `tags` will have the corresponding bit set:
+```
+{
+   "states": [
+     {
+       "tags": 1,
+       "start": 0,
+       "style": {
+         "parentKey": "string",
+         "values": {
+           "property1": {},
+           "property2": {}
+         }
+       },
+       "label": "string",
+       "end": 0
+     }
+   ]
+}
+```
+- When `tags` is set to 1 it is interpreted by the client as setting a dimmed opacity to the state. Similarly, when it is set to 4 it would be considered as being excluded. The reason for this bit mask is to minimize the amount of data transported per state. As previously mentioned, we should consider having `tags` correspond explicitly to either "DIMMED" or "EXCLUDED", although this might cause a slight effect on the performance.
+
+### Future Enhancements
+The initial implementation on the client will pass a single filter query(string) to the back-end and apply the changes. There is no feedback if the syntax is correct. However, there is a possibility to create **LSP (Language Server Protocol)**  in the back-end so that the front-end can use it to do syntax validation.
+
+## Decision
+
+TBD
```

Comment:
Here you should propose which solution to implement and the way to get there. Maybe describe increments if applicable.